### ISSUE TITLE
dotnet test CLI throws System.NotImplementedException when trying to load tests from a dll and passing runsettings arguments

### ISSUE THREAD

**Issue by Amiron49** [microsoft/vstest#2511]
## Description
the dotnet test cli fails while parsing any command targeting a .dll and passing runsettings commands. The same command works fine when providing a csproj or ommiting the target altogether

## Steps to reproduce
Running dotnet sdk version 3.1.302 and vstest cli version 16.6.0

The actual dll doesn't matter as the CLI crashes while parsing the arguments

```
dotnet test foo.dll -- anything
```

## Expected behavior

Not crash on parsing due to a NotImplementedException

https://github.com/microsoft/vstest/blob/da7967ac25841b4c259bdc36dbbc28a1cea56568/src/vstest.console/Processors/CLIRunSettingsArgumentProcessor.cs#L102

and probably yield the parsing of that arg to the method right next to it that can handle multiple parameters

https://github.com/microsoft/vstest/blob/da7967ac25841b4c259bdc36dbbc28a1cea56568/src/vstest.console/Processors/CLIRunSettingsArgumentProcessor.cs#L105

## Actual behavior
The cli crashes and throws a System.NotImplementedException

```
Microsoft (R) Testausführungs-Befehlszeilentool Version 16.6.0
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Unhandled exception. System.NotImplementedException: The method or operation is not implemented.
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.CLIRunSettingsArgumentExecutor.Initialize(String argument)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.ArgumentProcessorFactory.<>c__DisplayClass20_0.<WrapLazyProcessorToInitializeOnInstantiation>b__0()
   at System.Lazy`1.PublicationOnlyViaFactory(LazyHelper initializer)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.GetArgumentProcessors(String[] args, List`1& processors)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.Execute(String[] args)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Program.Main(String[] args)
```

## Diagnostic logs
[log.txt](https://github.com/microsoft/vstest/files/5035449/log.txt)

## Environment
Windows 10 1909

vstest.console 16.6.0

**Comment by nohwnd** [comment]
I think we fixed this while doing some parsing fixes and improvements to the -- settings syntax. Could you try to repro on net5.0? We would probably not bring the fix back to 3.1.x anyway. 

For me it does not repro: 

```
dotnet test UnitTestProject103.dll -- anything
Microsoft (R) Test Execution Command Line Tool Version 16.8.0-preview-20200730-02
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: 7 ms - UnitTestProject103.dll (net472)
```

**Comment by TrayanZapryanov** [comment]
Is there a chance this to be backported to 3.1 ? 
We are trying to use dotnet test in our release pipeline and we cannot pass any argument to the tests.....
I guess Azure will not be updated soon to support net5.0.

**Comment by Amiron49** [comment]
What I did end up doing to circumvent the issue was to apply a filter that would yield only the wanted assembly DLL

**Comment by nohwnd** [comment]
@TrayanZapryanov  quite unlikely, but you can download the net5.0 sdk (which is RC1) and use that to build netcorapp3.1 app and test it. The app that you produce will still require 3.1 runtime and will be able to run in the same places as before, you will just use a more modern set of tools to build it.

**Comment by TrayanZapryanov** [comment]
Thank you for suggestions. My case is Tfs Release pipeline step. There I can create step to execute dotnet test ... step with some unit testing over deployed newer version of our service in staging environment. From answers I understand that I have to wait until .net5 is integrated in pipelines.

**Comment by nohwnd** [comment]
There you should be able to use Use .NET Core step and install version 5.0.x (after checking Include Preview Versions). Right now 5.0.x resolves to 5.0.0-preview8 because that is the latest on preview channel. If you needed newer you would need to use the dotnet-install script directly.

![image](https://user-images.githubusercontent.com/5735905/93069525-0d705800-f67e-11ea-9c66-6e1848d0c439.png)

**Comment by TrayanZapryanov** [comment]
Thank you. I will try.
